### PR TITLE
Fix copy-constructor and copy-assignment operator in  `ContactPhaseList` class

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This Implements the copy-constructor and copy-assignment operator for the `ContactPhaseList` class. 

`ContactPhaseList` has an attribute called `m_phases`

https://github.com/dic-iit/bipedal-locomotion-framework/blob/d916b56f71eca03a7d1e519e4f13d785665b79ed/src/Contacts/include/BipedalLocomotion/Contacts/ContactPhaseList.h#L37

where `ContactPhase` has a `std::unordered_map` of active contacts during a given phase.

https://github.com/dic-iit/bipedal-locomotion-framework/blob/d916b56f71eca03a7d1e519e4f13d785665b79ed/src/Contacts/include/BipedalLocomotion/Contacts/ContactPhase.h#L37-L40

Accordingly to the `ContactPhaseList` documentation

https://github.com/dic-iit/bipedal-locomotion-framework/blob/d916b56f71eca03a7d1e519e4f13d785665b79ed/src/Contacts/include/BipedalLocomotion/Contacts/ContactPhaseList.h#L28-L31

all the iterators stored in `this->m_phases` should refer to `this->m_contactList`.
However when the default copy-constructor/copy-assignment operator are colled the content of `other.m_phases` is copied into `this->m_phases` and consequentially the iterators stored in `this->m_phases` will refer to `other.m_contactLists`.

I wrote an [ad-hoc test](https://github.com/dic-iit/bipedal-locomotion-framework/blob/5ec37f1d22af18f151e6c41541c37f795e2663bc/src/Contacts/tests/Contacts/ContactPhaseListTest.cpp#L15) to spot the bug. The test fails in Ubuntu 20.04 if Valgrind is used. Please check [here](https://github.com/dic-iit/bipedal-locomotion-framework/runs/2507414976).

----

In order to satisfy the rule of [five](https://en.cppreference.com/w/cpp/language/rule_of_three) the move-constructor move-assignment operator and destructor member functions must be defined.

----

I also introduced `TextLogging` in `ContactPhaseList`

Thank you @prashanthr05 for helping me in debugging 😃

### CODE REVIEW

File: src/Contacts/include/BipedalLocomotion/Contacts/ContactPhaseList.h [hunk 1]
```diff
@@ -43,6 +43,59 @@ class ContactPhaseList
...
+    ContactPhaseList(ContactPhaseList&& other) = default;
+
+    /**
+     * @brief Copy assignment operator.
+     * @param other another ContactPhaseList.
+     * @note When the copy assignment operator is called  the content of `other.m_contactList` will
+     * be copied in `this->m_contactList`, while `this->m_phases` will be regenerated using the
+     * content of `this->m_contactList`. Notice that Contacts::ContactPhase class stores an
+     * std::unordered_map<std::string, ContactList::const_iterator> called
+     * Contacts::ContactPhase::activeContacts. So coping the content of `other.m_phases` into
```
Comment (prashanthr05) [inline src/Contacts/include/BipedalLocomotion/Contacts/ContactPhaseList.h:86]:
```suggestion
     * Contacts::ContactPhase::activeContacts. So copying the content of `other.m_phases` into
```

File: src/Contacts/include/BipedalLocomotion/Contacts/ContactPhaseList.h [hunk 2]
```diff
@@ -43,6 +43,59 @@ class ContactPhaseList
...
+    ~ContactPhaseList() = default;
+
+    /**
+     * @brief Copy constructor.
+     * @param other another ContactPhaseList.
+     * @note When the copy constructor operator is called  the content of `other.m_contactList` will
+     * be copied in `this->m_contactList`, while `this->m_phases` will be regenerated using the
+     * content of `this->m_contactList`. Notice that Contacts::ContactPhase class stores an
+     * std::unordered_map<std::string, ContactList::const_iterator> called
+     * Contacts::ContactPhase::activeContacts. So coping the content of `other.m_phases` into
```
Comment (prashanthr05) [inline src/Contacts/include/BipedalLocomotion/Contacts/ContactPhaseList.h:66]:
```suggestion
     * Contacts::ContactPhase::activeContacts. So copying the content of `other.m_phases` into
```

File: src/Contacts/include/BipedalLocomotion/Contacts/ContactPhaseList.h [hunk 3]
```diff
@@ -43,6 +43,59 @@ class ContactPhaseList
...
+     * `this->m_phases` is wrong since all the iterators stored inside the contact phases should
+     * refer to the lists stored within this class, not the original input lists.
+     */
+    ContactPhaseList& operator=(const ContactPhaseList& other);
+
+    /**
+     * @brief Copy assignment operator.
```
Comment (prashanthr05) [inline src/Contacts/include/BipedalLocomotion/Contacts/ContactPhaseList.h:93]:
```suggestion
     * @brief Move assignment operator.
```

**Review by prashanthr05 — APPROVED** [review]
Minor comments.

**Comment by GiulioRomualdi** [comment]
@prashanthr05 I fixed the typos and I rebased

**Comment by prashanthr05** [comment]
Green light from me for the merge.

**Review by S-Dafarra — APPROVED** [review]
Thanks!

**Pull request merged**